sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/LogsIngestionAsyncClient.java

```diff
@@ -0,0 +1,268 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.monitor.ingestion;
+
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.http.rest.RequestOptions;
+import com.azure.core.models.ResponseError;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.Context;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.serializer.ObjectSerializer;
+import com.azure.monitor.ingestion.implementation.DefaultJsonSerializer;
+import com.azure.monitor.ingestion.implementation.IngestionUsingDataCollectionRulesAsyncClient;
+import com.azure.monitor.ingestion.implementation.UploadLogsResponseHolder;
+import com.azure.monitor.ingestion.models.UploadLogsError;
+import com.azure.monitor.ingestion.models.UploadLogsOptions;
+import com.azure.monitor.ingestion.models.UploadLogsResult;
+import com.azure.monitor.ingestion.models.UploadLogsStatus;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.zip.GZIPOutputStream;
+
+import static com.azure.core.util.FluxUtil.monoError;
+import static com.azure.core.util.FluxUtil.withContext;
+
+/**
+ * The asynchronous client for uploading logs to Azure Monitor.
+ */
+@ServiceClient(isAsync = true, builder = LogsIngestionClientBuilder.class)
+public final class LogsIngestionAsyncClient {
+    private static final ClientLogger LOGGER = new ClientLogger(LogsIngestionAsyncClient.class);
+    private static final String CONTENT_ENCODING = "Content-Encoding";
+    private static final long MAX_REQUEST_PAYLOAD_SIZE = 1024 * 1024; // 1 MB
+    private static final String GZIP = "gzip";
+    // TODO (srnagar): Move DefaultJsonSerializer in azure-core to public package
+    private static final DefaultJsonSerializer DEFAULT_SERIALIZER = new DefaultJsonSerializer();
```

Comment:
curious why do you need `DefaultJsonSerializer` to be public?

```suggestion
    private static final JsonSerializer DEFAULT_SERIALIZER = JsonSerializerProviders.createInstance(true);
```

Reply:
Oh right, we have the fallback method in the providers. Good catch!